### PR TITLE
Add additional remote sources build args

### PR DESCRIPTION
**Pull request by athos-ribeiro**
We add build arguments to allow remote sources to be copied into the
build image. Since the ADD command can extract tar archives directly,
there is no more need to extract the sources in the workdir.

To accomplish this, a Dockerfile must contain the following
instructions:

```
RUN mkdir $REMOTE_SOURCE_DIR
ADD $REMOTE_SOURCE $REMOTE_SOURCE_DIR
```

* OSBS-8473
* OSBS-8135

Signed-off-by: Athos Ribeiro <athos@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by athos-ribeiro** [comment]
Docs PR: https://github.com/athos-ribeiro/osbs-docs/pull/new/add-remote-source

File: atomic_reactor/plugins/pre_download_remote_source.py [hunk 1]
```diff
@@ -40,11 +41,19 @@ def run(self):
         # Download the source code archive
         archive = download_url(self.url, self.workflow.source.workdir)
 
-        # Unpack the source code archive into the workdir
```
Comment (athos-ribeiro) [inline atomic_reactor/plugins/pre_download_remote_source.py]: Read $COMMIT_MESSAGE for an explanation for removing this.

@rcerven brought an interesting question on IRC though: if we are to revert this and go back to extracting the sources in the workdir, would it be possible for files, such as the Dockerfile, to be overwritten? In this case, is this desirable?

**Comment by twaugh** [comment]
Do we need to watch out for https://github.com/openshift/imagebuilder/issues/99 and/or https://github.com/openshift/imagebuilder/issues/139 though?

File: atomic_reactor/plugins/pre_download_remote_source.py (hunk 1 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_download_remote_source.py]: I'd say that dockerfile shouldn't be replaced ever.

File: atomic_reactor/plugins/pre_download_remote_source.py [hunk 2]
```diff
@@ -40,11 +41,19 @@ def run(self):
...
-            tf.extractall(self.workflow.source.workdir)
+        if not tarfile.is_tarfile(archive):
+            raise RuntimeError('Remote source archive {} is not a tar file'.format(archive))
 
         # Set build args
         self.workflow.builder.buildargs.update(self.buildargs)
 
+        # To extract the sources into the build image, Dockerfile should contain
+        # RUN mkdir $REMOTE_SOURCE_DIR
+        # ADD $REMOTE_SOURCE $REMOTE_SOURCE_DIR
```
Comment (MartinBasti) [inline atomic_reactor/plugins/pre_download_remote_source.py]: https://github.com/openshift/imagebuilder/issues/99

Is this issue related?

**Comment by athos-ribeiro** [comment]
@twaugh, @MartinBasti: I did not account for those two imagebuilder issues. How about

- extract the sources in a temporary directory
- set $REMOTE_SOURCE to that directory (without a slash as the first char if in the workdir)
- advise COPY instead of ADD

?

**Comment by athos-ribeiro** [comment]
Release notes:

- when fetching cachito sources, set build args to allow users to copy sources to the build image
